### PR TITLE
Update example addon

### PR DESCRIPTION
**Pull request by patsy-issa**
This PR updates the following:

- Ember infrastructure
- Template colocation from classic component structure
- Update test

### CODE REVIEW

**Comment by patsy-issa** [comment]
Do we want to keep testing against `ember-canary` for this project?

**Comment by marcoow** [comment]
I think this is only an example testing against ember-release only should be sufficient.

File: examples/example-addon/addon/components/component-a/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Component from '@glimmer/component';
+
+export default class ComponentAComponent extends Component {
+}
```
Comment (marcoow) [inline examples/example-addon/addon/components/component-a/index.js:4]: I think this can be deleted?

**Review by marcoow — COMMENTED** [review]
I think it'd be great to use the component in the dummy app and have an acceptance test that ensures that works.

**Comment by patsy-issa** [comment]
Added acceptance test

**Review by marcoow — APPROVED** [review]

**Pull request merged**